packages/gestalt/src/TableRow.js

```diff
@@ -1,10 +1,41 @@
 // @flow strict
-import React, { type Node } from 'react';
+import React, { Children, type Node, cloneElement, useContext, useEffect, useState } from 'react';
+import TableContext from './TableContextProvider.js';
 
 type Props = {|
   children: Node,
 |};
 
 export default function TableRow(props: Props): Node {
-  return <tr>{props.children}</tr>;
+  const { stickyColumns } = useContext(TableContext);
+  const rowRef = React.useRef();
+  const [columnWidths, setColumnWidths] = useState([]);
+
+  useEffect(() => {
+    if (rowRef && rowRef.current && stickyColumns) {
+      const colWidths = [];
+      const tableRowChildrenArray = [...rowRef.current.children];
+      tableRowChildrenArray.forEach((child) => {
+        colWidths.push(child.clientWidth);
+      });
+      setColumnWidths(colWidths);
+    }
+  }, [rowRef, stickyColumns]);
+
+  const renderCellsWithIndex = () => {
+    const cells = [];
+    const tableRowChildrenArray = Children.toArray(props.children);
```

Comment:
This can be simplified with `Children.map`:
```js
const renderCellsWithIndex = () => Children.map(props.children, (item) => {
  ...
  return cloneElement(...);
});
```

Another idea to prevent needing this separate faux-render method would be to extract the map function to a helper, so the component return is something like
```js
  return <tr ref={rowRef}>{stickyColumns > 0 ? Children.map(props.children, renderCellsWithIndex) : props.children}</tr>;
```